tests/id/id.cpp

```diff
@@ -239,7 +239,8 @@ TEMPLATE_TEST_CASE_SIG("id can be implicitly conversion-constructed from item",
   q.submit([r, &result_buf](sycl::handler& cgh) {
      sycl::accessor result{result_buf, cgh, sycl::write_only};
      cgh.parallel_for<kernel_id<D>>(r, [=](sycl::item<D> itm) {
-       if (itm.get_id() == id<D>{r} - 1) {
+       // `ul` suffix is necessary to resolve ambiguity for id<1>
+       if (itm.get_id() == id<D>{r} - 1ul) {
```

Comment:
Why `l`?
Just `u` does not work?
The real type would be `uz` but it is C++23.

Reply:
Not sure, but `size_t` is at least `unsigned long` on all relevant architectures, no?